tests/test_signal_generator.cpp

```diff
@@ -0,0 +1,161 @@
+#include <hyro/SignalGeneratorComponent.h>
+#include <hyro/DigitalConverterComponent.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+#include <hyro/test/ComponentTester.h>
+
+#include <gtest/gtest.h>
+
+static void
+__assert_and_run_state_machine(hyro::StateMachine &sm,
+                               const hyro::ComponentConfiguration &conf,
+                               const hyro::ConnectionConfiguration &endpoint)
+{
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.init(conf));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.start());
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.connect(endpoint));
+
+  ASSERT_EQ(hyro::Result::RESULT_OK, sm.check());
+}
+
+#define ASSERT_AND_RUN_STATE_MACHINE(sm, conf, endpoint) \
+  __assert_and_run_state_machine(sm, conf, endpoint)
+
+using namespace signal_generator;
+using namespace digital_converter;
+
+namespace hyro
+{
+namespace test
+{
+
+TEST(ExampleTest, SignalGeneratorComponentCheck)
+{
+  //Configuration
+  auto generator_configuration = "{"
+                                 "outputs: {"
+                                 //"analog_output: { protocol: 'api', ip: '0.0.0.0',   port: '50051' },"
+                                 "analog_output: { protocol: 'api'},"
+                                 "fix_dynamic: { protocol: 'api'},"
+                                 "}"
+                                 "}";
+
+  //IntGeneratorComponent state machine
+  StateMachine sm(std::make_shared<SignalGeneratorComponent>("/generator"_uri));
+
+  auto result = sm.init(ComponentConfiguration(generator_configuration));
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.start();
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.connect(ConnectionConfiguration());
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  result = sm.check();
+  ASSERT_EQ(Result::RESULT_OK, result);
+
+  //Fake input channels
+  auto AnalogOut = std::make_shared<FakeInput<SignalMsgs>>("AnalogOut"_uri, "api", "/generator/analog_output");
+  ASSERT_TRUE(AnalogOut->connect());
+
+  DynamicPropertyAccess dynamic_property_access("/generator"_uri);
+
+  auto ret = dynamic_property_access.set<bool>("Cosine", true);
+  ASSERT_EQ(ret, true);
+  ASSERT_TRUE(dynamic_property_access.set<float>("Amplitude", 100.0));
+  ASSERT_TRUE(dynamic_property_access.set<float>("Frequency", 30.0));
+
+  bool cosine;
+  dynamic_property_access.get<bool>("Cosine", cosine);
+  ASSERT_EQ(cosine, true);
+
+  float amp, freq;
+  dynamic_property_access.get<float>("Amplitude", amp);
+  ASSERT_EQ(amp, 100.0);
+  dynamic_property_access.get<float>("Frequency", freq);
+  ASSERT_EQ(freq, 30.0);
+
+  sm.update();
+
+  auto ValueAnalogOut = std::shared_ptr<const SignalMsgs>();
```

Comment:
Please, use **snake_case** or **CamelCase** style. Do not declare the first letter of the variables as UpperCase.